src/main/java/com/faforever/client/replay/ReplayFileReaderImpl.java

```diff
@@ -34,16 +37,22 @@ public ReplayFileReaderImpl() {
   @SneakyThrows
   public LocalReplayInfo parseMetaData(Path replayFile) {
     logger.debug("Parsing metadata of replay file: {}", replayFile);
-    List<String> lines = Files.readAllLines(replayFile);
-    return gson.fromJson(lines.get(0), LocalReplayInfo.class);
+    Optional<String> metadata = Files.lines(replayFile).findFirst();
+    if (metadata.isPresent()) {
+      return gson.fromJson(metadata.get(), LocalReplayInfo.class);
+    }
+    throw new IOException(String.format("Failed to extract metadata from replay file: {}", replayFile));
```

Comment:
Instead of the if use  map(metadata -> gson.fromJson...).orElseThrow(new IOException...)
Also we should put Files.lines into a try-with-resources to ensure closing of the file

Reply:
Fixed, but I think readability suffers quite a bit with this style.